iroh-bytes/src/protocol/range_spec.rs

```diff
@@ -177,19 +214,23 @@ impl RangeSpecSeq {
         }
     }
 
-    /// An iterator over non empty range specs
+    /// An iterator over blobs in the collection with a non-emtpy range spec.
+    ///
+    /// This iterator will only yield items for blobs which have at least one chunk
     ///
-    /// This iterator is infinite if the range spec is infinite
+    /// This iterator is infinite if the [`RangeSpecSeq`] ends on a non-empty [`RangeSpec`],
+    /// that is all further blobs have selected chunks spans.
     pub fn iter_non_empty(&self) -> NonEmptyRequestRangeSpecIter<'_> {
         NonEmptyRequestRangeSpecIter::new(self.iter())
     }
 }
 
 static EMPTY_RANGE_SPEC: RangeSpec = RangeSpec::EMPTY;
 
-/// An infinite iterator of range specs
+/// An infinite iterator yielding [`RangeSpec`]s for each blob in a collection.
 ///
-/// default is what to use if the children of this RequestRangeSpec are empty.
+/// The first item yielded is the [`RangeSpec`] for the first blob in the collection, the
```

Comment:
This is not correct. The first item yielded is the RangeSpec for the collection itself.

This is needed e.g. when you already have the collection and don't want to download it again.

But as I mentioned above, I think it is best to leave out the notion of collections and just describe this as a sequence of `RangeSet`s.

Reply:
so i agree that "sequence of blobs" is probably a better description since when using "collection" you already are assuming special treatment of the first item/blob in the collection, which is not what I intended.  I fear "sequence of RangeSets" is a bit too abstract.

Reply:
So, how about this: we remove collection / children from both the docs and the API, and then add another section where we describe where a RangeSpecSeq is used in the case of collections and children (in a subsequent PR)?

I also would like to make the canonical repr mandatory. Without that you can't have structural equality. But better in the next PR.

Reply:
sounds good